src/main/java/org/apache/sysml/hops/AggUnaryOp.java

```diff
@@ -352,25 +359,41 @@ protected double computeIntermediateMemEstimate( long dim1, long dim2, long nnz
 				break;
 			case SUM:
 			case SUM_SQ:
-				//worst-case correction LASTROW / LASTCOLUMN 
+				//worst-case correction LASTROW / LASTCOLUMN
 				if( _direction == Direction.Col ) //(potentially sparse)
 					val = OptimizerUtils.estimateSizeExactSparsity(1, dim2, sparsity);
 				else if( _direction == Direction.Row ) //(always dense)
 					val = OptimizerUtils.estimateSizeExactSparsity(dim1, 1, 1.0);
 				break;
 			case MEAN:
-				//worst-case correction LASTTWOROWS / LASTTWOCOLUMNS 
+				//worst-case correction LASTTWOROWS / LASTTWOCOLUMNS
 				if( _direction == Direction.Col ) //(potentially sparse)
 					val = OptimizerUtils.estimateSizeExactSparsity(2, dim2, sparsity);
 				else if( _direction == Direction.Row ) //(always dense)
 					val = OptimizerUtils.estimateSizeExactSparsity(dim1, 2, 1.0);
 				break;
 			case VAR:
 				//worst-case correction LASTFOURROWS / LASTFOURCOLUMNS
-				if( _direction == Direction.Col ) //(potentially sparse)
+				if (DMLScript.USE_ACCELERATOR) {
+					// The GPU implementation only operates on dense data
+					// It allocated 2 dense blocks to help with these ops:
+					// it calculates the mean,
+					// subtracts the mean (row-wise or column-wise) from elements of the matrix into temp1
+					// it then squares temp1 to temp2
+					// it then divides all elements / each row / each column by the (size-1), (rowSize-1) or (colSize-1)
```

Comment:
The comment is confusing - you state 2 dense blocks, but you needs means, temp1, and temp2.